src/main/java/practice/StreamPractice.java

```diff
@@ -75,6 +91,25 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * parametrized with Candidate in CandidateValidator.
      */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        return candidates.stream()
+                .filter(new CandidateValidator())
+                .map(Candidate::getName)
+                .sorted()
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Given a List of `Person` instances (having `name`, `age`, `sex` and `cats` fields,
+     * and each `Cat` having a `name` and `age`),
+     * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
+     */
+    public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
+        return peopleList.stream()
+                .filter(w -> w.getSex() == Person.Sex.WOMAN && w.getAge() >= femaleAge
+                        && !w.getCats().isEmpty())
```

Comment:
```suggestion
                     )
```
по суті ця умова буде хендлетись через flatMap - якщо ліст пустий то просто в стрім нічого не додасться